server/lib/s3backup/README

```diff
@@ -0,0 +1,238 @@
+  of the tarball. The string is stored (and eventually returned on
+  inquiry) wrapped in double quotes followed by a newline.
+- an MD5 in file =<name>.MD5= (for "large" objects only) with no double
+  quote wrapping, followed by a newline.
+
+
+*** Mock implementation of getsize
+Mock =getsize()= treats the tarballs in a few selected tests as "large"
+and returns a size of 6 Gib for them. Otherwise, it returns the actual
+size of the tarball.
+
+*** Mock implementation of get_object
+
+=get_object()= returns a response that contains a small subset of the
+fields of the real response: the =ResponseMetadata= is a dict with one
+field, =HTTPStatusCode= with value 200, the =ETag= field contains the
+contents of the ETag file above (including double quotes, but no
+newline), and the =Metadata= field is either an empty dict ("small" objects)
+or contains two fields, =ETag= and =MD5= whose values are the contents
+of the corresponding file above ("large" objects).
+
+*** Mock implementation of list_objects
+=list_objects()= returns a response that contains two fields:
+=ResponseMetadata= (with =HTTPStatusCode= equal to 200) and =Contents=
+which is a list of entries, each of which contains a subset of the
+fields that the real =list_objects_v2()= returns: just =ETag=, =Size=
+and =Key=.  The =ETag= field contains the contents of the ETag file as
+above; the =Key= is basically "<controller>/<tb>.tar.xz"; the =Size=
+is set to 6GB for a "Large" object (one that has an MD5 file
+associated with it in the S3 "store") and the actual size of the
+tarball otherwise.
+
+One additional detail: if the bucket is large (i.e. it contains more
+than 1000 entries), the API returns only the first 1000, together with
+an indication that the result is truncated (=resp['IsTruncated']= is
+=True=) and a continuation token =resp[NextContinuationToken']=.
+=pbench-verify-backup-tarballs.py= calls =list_objects()= in a loop,
+setting its =ContinuationToken= to =resp['NextContinuationToken']=,
+indicating to the API that it wants the next 1000. The loop exits when
+the response does not contain a =NextContinuationToken= field (and the
+=IsTruncated= field is =False=, but we don't use that to determine
+when to exit: we assume that that is the case).
+
+ =list_objects()= implements a very simplified version of this
+behavior if the bucket name is ~SPECIAL_BUCKET~: that drives the mock
+=list_objects()= to treat the bucket as a large bucket, so the first
+time it gets called (without a =ContinuationToken=), it returns the
+first entry of however many it finds in the S3 "store", and sets the
+=IsTruncated= and =NextContinuationToken= fields in the response. The
+second time it is called (with a =ContinuationToken=), it returns the
+rest of the entries without a =NextContinuationToken=. It does not
+check for the "proper" continuation token (and it recalculates things
+inefficiently, but we don't worry about that).
+
+** Unit tests
```

Comment:
This was actually my scratch pad to jot down what each test was supposed to do. It really does not belong to the README file, so I should have deleted it, particularly since there are some questions that I did not answer. Do you think it's useful enough to put into unitttests, or should I just delete it?

Reply:
I deleted the section: I don't think it contained anything useful to move to unittests.